yt_dlp/extractor/videoken.py

```diff
@@ -0,0 +1,354 @@
+                break
+
+    def _real_extract(self, url):
+        hostname, category_id = self._match_valid_url(url).group('host', 'id')
+        org_id, _ = self._get_org_id_and_api_key(self._ORGANIZATIONS[hostname], category_id)
+        category = self._get_category_page(
+            category_id, org_id, note='Downloading category info')['category_name']
+        return self.playlist_result(
+            self._entries(category_id, org_id, url), category_id, category)
+
+
+class VideoKenTopicIE(VideoKenBaseIE):
+    _VALID_URL = VideoKenBaseIE._BASE_URL_RE + r'topic/(?P<id>[^/#?]+)/?(?:$|[?#])'
+    _TESTS = [{
+        'url': 'https://videos.neurips.cc/topic/machine%20learning/',
+        'playlist_mincount': 100,
+        'info_dict': {
+            'id': 'machine%20learning',
+            'title': 'machine learning',
+        },
+    }, {
+        'url': 'https://videos.icts.res.in/topic/gravitational%20waves/',
+        'playlist_mincount': 77,
+        'info_dict': {
+            'id': 'gravitational%20waves',
+            'title': 'gravitational waves'
+        },
+    }, {
+        'url': 'https://videos.cncf.io/topic/prometheus/',
+        'playlist_mincount': 134,
+        'info_dict': {
+            'id': 'prometheus',
+            'title': 'prometheus',
+        },
+    }]
+
+    def _get_topic_page(self, topic, org_id, search_id, api_key, page=1):
+        return self._download_json(
+            'https://es.videoken.com/api/v1.0/get_results', topic, fatal=False, query={
+                'orgid': org_id,
+                'size': '12',
+                'query': topic,
+                'page': page,
+                'sort': 'upload_desc',
+                'filter': 'all',
+                'token': api_key,
+                'is_topic': 'true',
+                'category': '',
+                'searchid': search_id,
+            }, headers={'Accept': 'application/json'}, note=f'Downloading topic page {page}') or {}
+
+    def _entries(self, topic, org_id, search_id, api_key, url):
+        for page in itertools.count(1):
+            videos = self._get_topic_page(topic, org_id, search_id, api_key, page)
+            total_pages = int_or_none(videos.get('total_no_of_pages'))
+            if not total_pages:
+                break
+            for video in traverse_obj(videos, ('results', ...)):
+                video_id = video.get('videoid')
+                if not video_id:
+                    continue
+                ie_key = None
+                if video.get('source') == 'youtube':
+                    video_url = video_id
+                    ie_key = 'Youtube'
+                else:
+                    video_url = video.get('embeddableurl')
+                    if urllib.parse.urlparse(video_url).netloc == 'slideslive.com':
+                        ie_key = 'SlidesLive'
+                        video_url = self._create_slideslive_url(video_url, video_id, url)
+                if not video_url:
+                    continue
```

Comment:
almost same code as single video. Can it be generalized?

Reply:
yeah I just had think like a smart person and `traverse_obj`